src/elements/TextInput/TextInput.tsx

```diff
@@ -103,12 +113,11 @@ export const TextInput: FC<TextInputProps> = ({
             setIsObfuscated(obfuscated);
         }
     }, [obfuscated]);
-
-    const onKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
-        if (event.key === "Enter") {
-            onEnterPressed && onEnterPressed((event.target as HTMLInputElement).value);
+    useEffect(() => {
+        if (inputElement.current && defaultValue) {
+            inputElement.current.value = defaultValue;
         }
-    };
```

Comment:
Any reason for the deletion?

Reply:
It's all been reverted, but the `TextInput` has become a really weird mix between a controlled and uncontrolled input. We'll probably have to revisit that at one point (or it might become something for the platform team in Q4).